Bluejay/Bluejay/Bluejay.swift

```diff
@@ -133,11 +136,13 @@ public class Bluejay: NSObject {
      
      - Parameters:
         - observer: An object interested in observing Bluetooth connection events and state changes. You can register more observers using the `register` function.
+        - showBluetoothAlert: Determines whether iOS will show a system alert when Bluetooth is turned off while the app using Bluejay is suspended. Defaults to false.
         - restoreMode: Determines whether Bluejay will opt-in to state restoration, and if so, can optionally provide a listen restorer as well for restoring listens.
         - coreBluetoothState: Allows starting Bluejay with an existing Core Bluetooth manager and peripheral.
     */
     public func start(
         connectionObserver observer: ConnectionObserver? = nil,
+        enableBluetoothAlert showBluetoothAlert: Bool = false,
         backgroundRestore restoreMode: BackgroundRestoreMode = .disable,
```

Comment:
This might be overthinking it, but it seems weird that in the case of the central manager options we are just directly including the properties that go into the options dictionary in the function call signature, even though it has basically the same sort of semantics as the ConnectionOptions, for which we are using a wrapper struct. Seems like if we wanted to be really consistent  enableBluetoothAlert and backgroundRestore should be in some "StartOptions" struct or something.

On the other hand, the main reason for the struct in connections is to enable setting the global defaults, which I don't think is as useful (maybe not at all useful) for initialization. If we didn't want to be able to do that global default, the connection options would probably be nicer as just parameters to the connect function as well. So it also seems bad to add extra abstraction here that is unnecessary and probably worse for users just to be consistent.

Not sure if I'm actually advocating for changing it or not, I think on balance I lean towards leaving it like it is now, but I'm unsure enough I thought it was worth raising as a possibility.

Reply:
I might not fully understand the problem raised here yet. Bluejay's `enableBluetoothAlert` param maps to CB's `CBCentralManagerOptionShowPowerAlertKey` and this pair is different from Bluejay's `ConnectionOptions` struct which maps to `CBConnectPeripheralOptionNotifyOnConnectionKey` or `CBConnectPeripheralOptionNotifyOnDisconnectionKey`. I think they do have different semantics, as we're comparing Bluetooth availability versus peripheral connection, so perhaps the distinction here is okay.

In fact, I think we might be more "correct" in the usage of CB by not specifying and dealing with neither `CBConnectPeripheralOptionNotifyOnConnectionKey` nor `CBConnectPeripheralOptionNotifyOnDisconnectionKey` when initializing the manager, as the scope at that point doesn't really involve any connections just yet. And if there are connections to restore, it should already have the options cached with the connection that' about to be restored.

Reply:
I'm not saying that the options should be combined in any way, or that we should be passing the connection options into the start struct. 

All I was pointing out is that we have one set of options (the connection options) that are passed in as a struct and then turned into a dictionary to pass to CoreBluetooth, and another set that are passed in as parameters to a function then turned into a dictionary object to pass to CB.

I'm speculating that maybe the signature for start should be something like:

```
struct StartOptions {
    var enableBluetoothAlert showBluetoothAlert: Bool = false,
    var backgroundRestore restoreMode: BackgroundRestoreMode = .disable
}

func start(connectionObserver observer: ConnectionObserver? = nil, options: StartOptions? = nil, coreBluetoothState: (manager: CBCentralManager, peripheral: CBPeripheral?)? = nil)
```

So that it follows the same approach as connection, the options that are going to be turned into the option dictionary to get passed to CoreBluetooth at manager initialization are wrapped up. In a struct.

Like I said, I'm not sure if this is actually better, it's just a little more consistent with the way we deal with options in connection.

In fact, taking it a little further, we should actually be restricted to EITHER specifying the start options or an existing CB state, since those two are actually mutually exclusive.

```
struct StartOptions {
    var enableBluetoothAlert showBluetoothAlert: Bool = false,
    var backgroundRestore restoreMode: BackgroundRestoreMode = .disable
}

enum StartMode {
    case create(StartOptions?)
    case useExisting(manager: CBCentralManager, peripheral: CBPeripheral?)
}

func start(connectionObserver observer: ConnectionObserver? = nil, mode: StartMode)

```